lib/actions-oclif/device/init.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, stripIndent } from '../../utils/lazy';
+import { runCommand } from '../../utils/helpers';
+
+interface FlagsDef {
+	application?: string;
+	app?: string;
+	yes: boolean;
+	advanced: boolean;
+	'os-version'?: string;
+	drive?: string;
+	config?: string;
+	help: void;
+}
+
+export default class DeviceInitCmd extends Command {
+	public static description = stripIndent`
+		Initialise a device with balenaOS.
+
+		Initialise a device by downloading the OS image of a certain application
+		and writing it to an SD Card.
+
+		Note, if the application option is omitted it will be prompted
+		for interactively.
+		`;
+	public static examples = [
+		'$ balena device init',
+		'$ balena device init --application MyApp',
+	];
+
+	public static usage = 'device init';
+
+	public static flags: flags.Input<FlagsDef> = {
+		application: cf.application,
+		app: cf.app,
+		yes: cf.yes,
+		advanced: flags.boolean({
+			char: 'v',
+			description: 'show advanced configuration options',
+		}),
+		'os-version': flags.string({
+			description: stripIndent`
+				exact version number, or a valid semver range,
+				or 'latest' (includes pre-releases),
+				or 'default' (excludes pre-releases if at least one stable version is available),
+				or 'recommended' (excludes pre-releases, will fail if only pre-release versions are available),
+				or 'menu' (will show the interactive menu)
+				`,
+		}),
+		drive: flags.string({
+			char: 'd',
+			description: stripIndent`
+				the drive to write the image to, eg. \`/dev/sdb\` or \`/dev/mmcblk0\`. \
+				Careful with this as you can erase your hard drive. \
+				Check \`balena util available-drives\` for available options.
+				`,
+		}),
+		config: flags.string({
+			description: 'path to the config JSON file, see `balena os build-config`',
+		}),
+		help: cf.help,
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { flags: options } = this.parse<FlagsDef, {}>(DeviceInitCmd);
+
+		// Imports
+		const { promisify } = await import('bluebird');
```

Comment:
Maybe `import { promisify } from 'util';` like Page did for example here:
https://github.com/balena-io/balena-cli/pull/1907/files#diff-f49db67e2f5f28913d23492aec7161d2R20

Reply:
Just tried this, but it doesn't seem to work as a drop in replacement - I'm getting typescript compilation errors regarding tmpNameAsync.